File: src/lib/api/videoIngressClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class VideoIngress {
  constructor(private readonly host: RPCHost) {}

  public isSupported(
    arg: pb.IVideoIngressIsSupportedRequest = new pb.VideoIngressIsSupportedRequest()
  ): Promise<pb.VideoIngressIsSupportedResponse> {
    return this.host.expectOne(
      this.host.call("VideoIngress/IsSupported", new pb.VideoIngressIsSupportedRequest(arg))
    );
  }
  public getConfig(
    arg: pb.IVideoIngressGetConfigRequest = new pb.VideoIngressGetConfigRequest()
  ): Promise<pb.VideoIngressGetConfigResponse> {
    return this.host.expectOne(
      this.host.call("VideoIngress/GetConfig", new pb.VideoIngressGetConfigRequest(arg))
    );
  }
  public setConfig(
    arg: pb.IVideoIngressSetConfigRequest = new pb.VideoIngressSetConfigRequest()
  ): Promise<pb.VideoIngressSetConfigResponse> {
    return this.host.expectOne(
      this.host.call("VideoIngress/SetConfig", new pb.VideoIngressSetConfigRequest(arg))
    );
  }
  public listStreams(
    arg: pb.IVideoIngressListStreamsRequest = new pb.VideoIngressListStreamsRequest()
  ): Promise<pb.VideoIngressListStreamsResponse> {
    return this.host.expectOne(
      this.host.call("VideoIngress/ListStreams", new pb.VideoIngressListStreamsRequest(arg))
    );
  }
  public getChannelURL(
    arg: pb.IVideoIngressGetChannelURLRequest = new pb.VideoIngressGetChannelURLRequest()
  ): Promise<pb.VideoIngressGetChannelURLResponse> {
    return this.host.expectOne(
      this.host.call("VideoIngress/GetChannelURL", new pb.VideoIngressGetChannelURLRequest(arg))
    );
  }
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/VideoIngressClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class VideoIngressClient(filepath: String) : RPCClient(filepath) {

    suspend fun isSupported(
        arg: VideoIngressIsSupportedRequest = VideoIngressIsSupportedRequest()
    ): VideoIngressIsSupportedResponse =
        this.callUnary("VideoIngress/IsSupported", arg)

    suspend fun getConfig(
        arg: VideoIngressGetConfigRequest = VideoIngressGetConfigRequest()
    ): VideoIngressGetConfigResponse =
        this.callUnary("VideoIngress/GetConfig", arg)

    suspend fun setConfig(
        arg: VideoIngressSetConfigRequest = VideoIngressSetConfigRequest()
    ): VideoIngressSetConfigResponse =
        this.callUnary("VideoIngress/SetConfig", arg)

    suspend fun listStreams(
        arg: VideoIngressListStreamsRequest = VideoIngressListStreamsRequest()
    ): VideoIngressListStreamsResponse =
        this.callUnary("VideoIngress/ListStreams", arg)

    suspend fun getChannelURL(
        arg: VideoIngressGetChannelURLRequest = VideoIngressGetChannelURLRequest()
    ): VideoIngressGetChannelURLResponse =
        this.callUnary("VideoIngress/GetChannelURL", arg)

}
